src/net/http_serde.rs

```diff
@@ -135,7 +135,29 @@ pub mod query {
                         num_multi_bits,
                     }) = req.extract().await?;
 
-                    Ok(QueryType::Ipa(IpaQueryConfig {
+                    Ok(QueryType::SemiHonestIpa(IpaQueryConfig {
+                        per_user_credit_cap,
+                        max_breakdown_key,
+                        attribution_window_seconds,
+                        num_multi_bits,
+                    }))
+                }
+                QueryType::MALICIOUS_IPA_STR => {
+                    #[derive(serde::Deserialize)]
+                    struct IPAQueryConfigParam {
+                        per_user_credit_cap: u32,
+                        max_breakdown_key: u32,
+                        attribution_window_seconds: Option<NonZeroU32>,
+                        num_multi_bits: u32,
+                    }
+                    let Query(IPAQueryConfigParam {
+                        per_user_credit_cap,
+                        max_breakdown_key,
+                        attribution_window_seconds,
+                        num_multi_bits,
+                    }) = req.extract().await?;
```

Comment:
Can this be combined like for semihonest?

Reply:
I'm not sure I follow what you mean.  These appear to also be separate from semihonest in lines 125-136.  what am I missing?

Reply:
okay, updated this to remove some of the duplicate code in the match.